### ISSUE TITLE
Adding loss view during training and feeding tensorboard - [merged]

### ISSUE THREAD

**Issue by srcansiz** [fedbiomed/fedbiomed#563]
_Merges feature/108-loss-view-researcher -> develop_

- HistoryLogger naming revisions
- Extra keys has been added to AddScalarReply dataclass 
- add_scalar method has been activated for training routine
- Monitoring class is created for researcher side that independently listens general/monitoring channel during training routine
- New messaging type is added for messaging.py 
- Loss view and tensorboard for PyTorch implementation
- Loss view and tensorboard for Scikit-learn, using SGDRegressor, SGDClassier and Perceptron

Close #108
